### PR TITLE
1419 apache on server

### PR DESCRIPTION
**Pull request by mlissner**

### CODE REVIEW

File: docker/apache/sites-available/ssl-courtlistener.com.conf [hunk 1]
```diff
@@ -9,13 +11,11 @@
     # Since Let's Encrypt can auto-update some of this stuff it just seems
     # safer to repeat it.
     SSLEngine on
-    SSLProtocol all -SSLv2 -SSLv3
-    SSLCipherSuite "kEECDH+ECDSA+AES128 kEECDH+ECDSA+AES256 kEECDH+AES128 kEECDH+AES256 kEDH+AES128 kEDH+AES256 DES-CBC3-SHA +SHA !aNULL !eNULL !LOW !MD5 !EXP !DSS !PSK !SRP !kECDH !CAMELLIA !RC4 !SEED"
-    SSLHonorCipherOrder on
+    SSLProtocol            all -SSLv2 -SSLv3 -TLSv1 -TLSv1.1
```
Comment (drewsilcock) [inline docker/apache/sites-available/ssl-courtlistener.com.conf:14]: Do you still need to support SSLv2? It's not recommended due to downgrade attacks and other security flaws.

p.s. have you seen Mozilla tool for up-to-date secure configs for various servers? https://ssl-config.mozilla.org/#server=apache&version=2.4.41&config=intermediate&openssl=1.1.1d&guideline=5.6

Comment (drewsilcock) [reply docker/apache/sites-available/ssl-courtlistener.com.conf:14]: The other thing that the Mozilla tool mentions is setting `SSLSessionTickets off` which ([apparently](https://httpd.apache.org/docs/trunk/ssl/ssl_howto.html)) ensures forward secrecy.

There's also OCSP stapling which seems to mean that the browser doesn't need to contact the CA to check that the certificate hasn't been revoked. Enabling stapling can reduce the TLS overhead by about a third ([apparently](https://blog.cloudflare.com/ocsp-stapling-how-cloudflare-just-made-ssl-30/)).

**Comment by drewsilcock** [comment]
If you're running Apache using Docker instead of as a service, are you planning on creating a service to ensure the Apache Docker image gets run automatically?

Also you're probably aware of this but you have to be careful with symlinks and Docker containers - you can't just mount a directory with a symlink in it if the symlink points outside that directory.

File: docker/apache/sites-available/ssl-courtlistener.com.conf (hunk 1 above)
Comment (mlissner) [reply docker/apache/sites-available/ssl-courtlistener.com.conf:14]: > `SSLSessionTickets`

I used the mozilla configurator, but missed the `SSLSessionTickets` directive. Thanks for catching that. Fixed.

> `SSLv2`

The "Intermediate" setting for the configurator doesn't ban SSLv2, so I'm just rolling with that. That said, once this is up, I think I'll point the SSLLabs test thing at it, and if it tells me to block more protocols, I probably will.

> OCSP stapling

I don't know much about OCSP stapling. I'm in favor of it generally, but I haven't done the research to learn if it has any downsides. Doesn't seem like it? That CloudFlare thing is eight years old! Do you know though if it's just safe to turn it on with no downsides?

**Comment by mlissner** [comment]
> are you planning on creating a service to ensure the Apache Docker image gets run automatically?

I was going to use a docker compose file? IIRC, they have auto-restart stuff, right? I haven't looked at it in a year or so, but this is what we use for Solr and Celery already. 

> Also you're probably aware of this but you have to be careful with symlinks and Docker containers - you can't just mount a directory with a symlink in it if the symlink points outside that directory.

Ugh. I did *not* know that. I would have caught that during testing, but ugh, thanks for the heads up.

File: docker/apache/sites-available/ssl-courtlistener.com.conf (hunk 1 above)
Comment (drewsilcock) [reply docker/apache/sites-available/ssl-courtlistener.com.conf:14]: Both the "intermediate" and "old" settings leave out SSLv2 when I look at it.

I'm not sure whether OCSP stapling has any downsides tbh, I just found out about it from that Mozilla tool and some Googling around. It defaults to ticked in the SSL configuration generator, anyway.

**Comment by drewsilcock** [comment]
Yeah docker-compose has auto-restart which is a perfectly legitimate way of doing it I think.

**Pull request merged**